components/molecules/HighlightInput/highlight-input-form.tsx

```diff
@@ -280,12 +321,13 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
           newHighlightSuggestions.push(suggestionPage);
         }
 
+        await removeAlreadyHighlightedSuggestions(newHighlightSuggestions);
         setHighlightSuggestions(newHighlightSuggestions);
```

Comment:
```js
await removeAlreadyHighlightedSuggestions(newHighlightSuggestions); // <- setHighlightSuggestions(newHighlightSuggestions)
setHighlightSuggestions(newHighlightSuggestions); // setting the state again.
```

The `await removeAlreadyHighlightedSuggestions(newHighlightSuggestions);` function already sets the state of the `highlightSuggestions` within, but you are once again setting the state in next line `setHighlightSuggestions(newHighlightSuggestions);`

If you let the `removeAlreadyHighlightedSuggestions` return a filtered array of un-highlighted suggestions, you will have something some readable like so...

```js
const filteredSuggestions = await removeAlreadyHighlightedSuggestions(newHighlightSuggestions);
setHighlightSuggestions(filteredSuggestions);
```